_source/_posts/2021-07-28-toggle-feature-flags-csharp.md

```diff
@@ -0,0 +1,222 @@
+
+## How to Implement Feature Flags in .NET Core
+
+Now, I will demonstrate how to implement and use feature flags in .NET Core using a sample application created with the Okta CLI. For managing and accessing feature flags from my code, I'm going to use the [ConfigCat SDK for .NET](https://www.configcat.com/docs/sdk-reference/csharp).
+
+In this example, my new feature will be Okta's Twitter feed, embedded to the home page of the application. The feed should only be visible if its feature flag is turned on.
+
+## Before You Get Started
+
+### Set Up a Sample Application with the Okta CLI
+
+Okta is a secure and customizable solution to add authentication to your app. Okta supports any language on any stack and lets you define how you want your users to sign in. Each time a user tries to authenticate, Okta will verify his/her identity and send the required information back to your app.
+
+The [Okta CLI](https://github.com/okta/okta-cli) provides turnkey application templates with configured login flows.
+
+Install the [Okta CLI](https://github.com/okta/okta-cli) by following the steps on the [GitHub page](https://github.com/okta/okta-cli).
+
+If you already have an [Okta Developer](https://developer.okta.com/signup) account, run the following command:
+
+```sh
+okta login
+```
+
+If you don't, you can register a new account by running:
+
+```sh
+okta register
+```
+
+The Okta CLI is now set up and ready! Run:
+
+```sh
+okta start
+```
+
+to launch the CLI.
+
+Select the `ASP.NET Core MVC + Okta` option when prompted.
+
+A ready application will be created with the Okta login flow, enabling you to write secrets with `appsettings.json` or to add the new client to your Applications on the [Okta Developer Console](https://developer.okta.com).
+
+You can test run the application by hitting `F5`or by entering `dotnet run` in your command line.
+A browser window will open, and you should be able to log in using your Okta credentials.
+
+I have uploaded my working application to GitHub in case you get stuck: [Okta .NET Core 3 Feature Flag Example](https://github.com/mr-sige/okta-netcore3-feature-flags).
+
+### Set Up a ConfigCat Account
+
+Go to [ConfigCat.com](https://www.configcat.com) and sign up for a free account.
+
+You will see your first feature flag `My awesome feature flag` already created.
+
+{% img blog/toggle-feature-flags-csharp/image1.png alt:"" width:"800" %}{: .center-image }
+
+You can work with this feature flag, but I'm going to add a different one for my Twitter feed feature.
+
+{% img blog/toggle-feature-flags-csharp/image2.png alt:"" width:"800" %}{: .center-image }
```

Comment:
Alt tags would be great for SEO and accessiblity

Reply:
I agree. I will work with contractors on providing that in the future.